### ISSUE TITLE
Different behavior on linux/windows when wrapper element has same name as list element

### ISSUE THREAD

**Issue by vmodrosedem** [dmak/jaxb-xew-plugin#74]
I have stumbled on a bug in jaxb-xew-plugin, that leads to different code being generated when the plugin is run on linux and windows machines.
It happens in a specific case, when the XSD contains a list with a wrapper element and the wrapper and list elements have the same name. E.g.: 
```xml
<xsd:element name="RootElement">
    <xsd:complexType>
        <xsd:sequence>
            <xsd:element name="TestList" minOccurs="0">
                <xsd:complexType>
                    <xsd:sequence>
                        <xsd:element name="TestList" maxOccurs="unbounded">
                            <xsd:complexType>
                                <xsd:sequence>
                                    <xsd:element name="SomeField" type="xsd:string"/>
                                </xsd:sequence>
                            </xsd:complexType>
                        </xsd:element>
                    </xsd:sequence>
                </xsd:complexType>
            </xsd:element>
        </xsd:sequence>
    </xsd:complexType>
</xsd:element>
```

When run on windows, the generated jaxb RootElement class has a list field, that looks like this:
```java
@XmlElementWrapper(name = "TestList")
@XmlElement(name = "TestList", namespace = "...")
protected List<RootElement.TestListTestList> testList = new ArrayList<RootElement.TestListTestList>();
```
, but when run on linux, a different signature is generated:
```java
@XmlElementWrapper(name = "TestList")
@XmlElement(name = "TestList", namespace = "...")
protected List<RootElement.TestList> testList = new ArrayList<RootElement.TestList>();
```
The names of the list element class are different, i.e. RootElement.TestListTestList on windows vs RootElement.TestList on linux.

It seems, that [this block](https://github.com/dmak/jaxb-xew-plugin/blob/f2320aaa24d70d087ff24edc326912ac22d1acb8/src/main/java/com/sun/tools/xjc/addon/xew/XmlElementWrapperPlugin.java#L907C1-L910C4), that renames the class when the name is in conflict with an existing class, does not work correctly when running on a case-sensitive filesystem. That's because on a case-sensitive filesystem, the JDefinedClass#classes has class names in uppercase and then `classes.containsKey(clazz.name())` always returns false.

I have prepared a test example: [xew-test-case.zip](https://github.com/dmak/jaxb-xew-plugin/files/12381662/xew-test-case.zip), that works fine on windows, but fails to compile on linux.

**Comment by dmak** [comment]
Indeed there is something strage going. Windows log (compiles fine):
```
[INFO] Candidates:
[INFO] 	[+]: xew.jbo.cz.xew_test_case.RootElement.TestList
[INFO] 	1 candidate(s) being considered.
[INFO] 
[INFO] Modifications:
[INFO] 	Replacing field [TestList xew.jbo.cz.xew_test_case.RootElement#testList]
[INFO] 	Moving inner class xew.jbo.cz.xew_test_case.RootElement.TestList.TestList to class xew.jbo.cz.xew_test_case.RootElement
[INFO] 	Renaming class xew.jbo.cz.xew_test_case.RootElement.TestList.TestList to class TestListTestList
[INFO] 	Renamed createRootElementTestListTestList -> createRootElementTestListTestList in xew.jbo.cz.xew_test_case.ObjectFactory
[INFO] 	2 modification(s) to original code.
[INFO] 
[INFO] Deletions:
[INFO] 	Removing class xew.jbo.cz.xew_test_case.RootElement.TestList from class xew.jbo.cz.xew_test_case.RootElement
[INFO] 	Removing factory method [xew.jbo.cz.xew_test_case.RootElement.TestList#createRootElementTestList()] from xew.jbo.cz.xew_test_case.ObjectFactory
[INFO] 	2 deletion(s) from original code.
```
Linux log (does not compile):
```
[INFO] Candidates:
[INFO] 	[+]: xew.jbo.cz.xew_test_case.RootElement.TestList
[INFO] 	1 candidate(s) being considered.
[INFO] 
[INFO] Modifications:
[INFO] 	Replacing field [TestList xew.jbo.cz.xew_test_case.RootElement#testList]
[INFO] 	Moving inner class xew.jbo.cz.xew_test_case.RootElement.TestList.TestList to class xew.jbo.cz.xew_test_case.RootElement
[INFO] 	Renamed createRootElementTestListTestList -> createRootElementTestList in xew.jbo.cz.xew_test_case.ObjectFactory
[INFO] 	2 modification(s) to original code.
[INFO] 
[INFO] Deletions:
[INFO] 	Removing class xew.jbo.cz.xew_test_case.RootElement.TestList from class xew.jbo.cz.xew_test_case.RootElement
[INFO] 	Removing factory method [xew.jbo.cz.xew_test_case.RootElement.TestList#createRootElementTestList()] from xew.jbo.cz.xew_test_case.ObjectFactory
[INFO] 	2 deletion(s) from original code.
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /mnt/fantec/Backups/EPO/xew-test-case/src/main/java/cz/jbo/xew/Main.java:[10,26] cannot find symbol
  symbol:   class TestListTestList
  location: class xew.jbo.cz.xew_test_case.RootElement
[ERROR] /mnt/fantec/Backups/EPO/xew-test-case/src/main/java/cz/jbo/xew/Main.java:[10,73] cannot find symbol
  symbol:   class TestListTestList
  location: class xew.jbo.cz.xew_test_case.RootElement
```

**Comment by dmak** [comment]
The issue should be fixed, try [jaxb-xew-plugin-2.2-20230820.234133-1.jar](https://oss.sonatype.org/content/repositories/snapshots/com/github/jaxb-xew-plugin/jaxb-xew-plugin/2.2-SNAPSHOT/jaxb-xew-plugin-2.2-20230820.234133-1.jar). As a temporary measure you could replace released jar with that one. True release will happen somewhen later this year, let me know if that is burning for you.
